### PR TITLE
[Workers] Wrangler: Document keep_vars, and how wrangler.toml is a source of truth 

### PR DESCRIPTION
**Pull request by rozenmd**
We hotfixed `keep_vars` into Wrangler2 last night (https://github.com/cloudflare/wrangler2/pull/1895/files#diff-ff9d7ebee3125a28b76b0d0698fbe1021d1ebd0a6f0fe55253060001a7999751), adding docs for it in this PR.

### CODE REVIEW

File: content/workers/wrangler/configuration.md [hunk 1]
```diff
@@ -6,11 +6,13 @@ weight: 3
 
 # Configure `wrangler.toml`
 
-Wrangler optionally uses a `wrangler.toml` configuration file to customize the development and publishing setup for a Worker. This document serves as a reference for all the fields and acceptable values in this configuration file.
+Wrangler optionally uses a `wrangler.toml` configuration file to customize the development and publishing setup for a Worker. By default, `wrangler.toml` is the [source of truth](#source-of-truth) for configuring a Worker.
```
Comment (petebacondarwin) [inline content/workers/wrangler/configuration.md]: I am not sure this is strictly true. Perhaps we could relax the statement a bit to:

> It is best practice to treat `wrangler.toml` as the [source of truth](#source-of-truth) for configuring a Worker.

File: content/workers/wrangler/configuration.md [hunk 2]
```diff
@@ -51,18 +53,23 @@ routes = ["routes"] | [{ pattern = "*", zone_id = "ZONE_ID" }] | [{ pattern = "*
...
+# If your development approach is to modify vars after deployment via the dashboard you may wish to enable this option.
+keep_vars = true
 
 # A map of environment variables to set when deploying your Worker.
 # @default `{}`
 # not inherited
 [vars]
 KEY = "value"
 
+# Lets you call Workers periodically, much like a cron job.
```
Comment (petebacondarwin) [inline content/workers/wrangler/configuration.md]: Not strictly relevant to this PR but I would say:

```suggestion
# Lets you trigger Workers periodically, much like a cron job.
```

File: content/workers/wrangler/configuration.md [hunk 3]
```diff
@@ -261,3 +268,11 @@ header: .dev.vars
 ---
 SECRET_KEY = "value"
 ```
+
+## Source of truth
+
+By default, `wrangler.toml` serves as a source of truth for your Worker configuration. This allows you to treat `wrangler.toml` as a form of Infrastructure as Code.
```
Comment (petebacondarwin) [inline content/workers/wrangler/configuration.md]: Again, I am not sure we can say this. Strictly the source of truth is whatever is currently deployed on EWC. Developers can, and often do, make changes to their Worker via the dashboard (or even via custom scripts that hit EWC directly). Moreover, secrets (encrypted vars) cannot be specified at all in wrangler.toml right now.

I think we should say that it is recommended practice to use wrangler.toml as the source of truth and avoid making changes via other methods if possible.

Comment (rozenmd) [reply content/workers/wrangler/configuration.md]: Right, I see what you mean

File: content/workers/wrangler/configuration.md [hunk 4]
```diff
@@ -261,3 +268,11 @@ header: .dev.vars
 ---
 SECRET_KEY = "value"
 ```
+
+## Source of truth
+
+It is a recommended best-practice to treat `wrangler.toml` as a source of truth for your Worker configuration, and avoid making changes via the Cloudflare dashboard. This allows you to treat `wrangler.toml` as a form of Infrastructure as Code.
```
Comment (deadlypants1973) [inline content/workers/wrangler/configuration.md]:
```suggestion
It is a recommended best practice to treat `wrangler.toml` as a source of truth for your Worker configuration, and avoid making changes via the Cloudflare dashboard. This allows you to treat `wrangler.toml` as a form of Infrastructure as Code.
```

File: content/workers/wrangler/configuration.md [hunk 5]
```diff
@@ -261,3 +268,11 @@ header: .dev.vars
...
+
+If you change your environment variables in the Cloudflare dashboard, Wrangler _will_ override them the next time you deploy. If you want to disable this behavior, add `keep_vars = true` to your `wrangler.toml`.
```
Comment (deadlypants1973) [inline content/workers/wrangler/configuration.md]:
```suggestion
If you change your environment variables in the Cloudflare dashboard, Wrangler will override them the next time you deploy. If you want to disable this behavior, add `keep_vars = true` to your `wrangler.toml`. 
```

File: content/workers/wrangler/configuration.md [hunk 6]
```diff
@@ -261,3 +268,11 @@ header: .dev.vars
...
 SECRET_KEY = "value"
 ```
+
+## Source of truth
+
+It is a recommended best-practice to treat `wrangler.toml` as a source of truth for your Worker configuration, and avoid making changes via the Cloudflare dashboard. This allows you to treat `wrangler.toml` as a form of Infrastructure as Code.
+
+If you change your environment variables in the Cloudflare dashboard, Wrangler _will_ override them the next time you deploy. If you want to disable this behavior, add `keep_vars = true` to your `wrangler.toml`. 
+
+Note that wrangler won't delete your secrets (encrypted environment variables) unless you run `wrangler secret delete <key>`.
```
Comment (deadlypants1973) [inline content/workers/wrangler/configuration.md]:
```suggestion
Note that Wrangler will not delete your secrets (encrypted environment variables) unless you run `wrangler secret delete <key>`.
```

**Review by deadlypants1973 — APPROVED** [review]

**Pull request merged**